### PR TITLE
Runtime language switching + FR translation update

### PR DESCRIPTION
**Pull request by adriweb**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -81,10 +81,6 @@ int main(int argc, char **argv)
         app.setFont(QFont(QStringLiteral("Helvetica Neue")));
     #endif
 
-    QTranslator appTranslator;
```
Comment (Vogtinator) [inline main.cpp]: Moving this away breaks translations in the mobile UI where there is no MainWindow instance.

Comment (adriweb) [reply main.cpp]: Hmm, I'll put some things under the ifdef

**Comment by adriweb** [comment]
Tested the mobile UI now, works fine (it's translated as expected) on a French system for instance.

**Comment by Vogtinator** [comment]
Switching doesn't seem to have an effect on the configuration dialog's sidebar.
Not sure why, normally ListElement should support that...

Feature request: Save the language if the user explicitly set one (otherwise keep the system default)

**Comment by adriweb** [comment]
Indeed, the sidebar of the config doesn't get translated (until relaunch anyway), that's weird (probably a bug?).

Regarding the saving feature, it's already the case, actually :)

File: mainwindow.cpp [hunk 2]
```diff
@@ -224,6 +238,40 @@ MainWindow::~MainWindow()
...
+    {
+        settings->setValue(QStringLiteral("prefered_lang"), lang);
+    } else {
+        QMessageBox::warning(this, tr("Language change"), tr("No translation available for this language :("));
+    }
+}
+
+void MainWindow::changeEvent(QEvent* event)
+{
+    if (event)
```
Comment (Vogtinator) [inline mainwindow.cpp]: This won't be `nullptr`.

File: mainwindow.cpp [hunk 3]
```diff
@@ -224,6 +238,40 @@ MainWindow::~MainWindow()
...
+    {
+        switch(event->type())
+        {
+            case QEvent::LanguageChange:
+                ui->retranslateUi(this);
+                break;
+
+            case QEvent::LocaleChange:
+                switchTranslator(QLocale::system().name());
```
Comment (Vogtinator) [inline mainwindow.cpp]: Add a check for `preferred_language == none`.

File: mainwindow.cpp [hunk 4]
```diff
@@ -224,6 +238,40 @@ MainWindow::~MainWindow()
...
+                break;
+
+            default:
+                break;
```
Comment (Vogtinator) [inline mainwindow.cpp]: In this case I'd prefer a `if`/`else if` chain to a `switch`.

File: mainwindow.h [hunk 5]
```diff
@@ -131,13 +132,17 @@ public slots:
 
     void refillKitMenus();
 
+    void switchTranslator(const QString& lang);
```
Comment (Vogtinator) [inline mainwindow.h]: Needs a comment to explain what it does - it's not obvious that it changes the settings.

File: mainwindow.ui [hunk 6]
```diff
@@ -655,6 +665,21 @@
     <string>Switch to Mobile UI</string>
    </property>
   </action>
+  <action name="actionEnglish">
```
Comment (Vogtinator) [inline mainwindow.ui]: Sort alphabetically please.

File: mainwindow.cpp [hunk 7]
```diff
@@ -33,6 +33,10 @@ MainWindow::MainWindow(QWidget *parent) :
     QMainWindow(parent),
     ui(new Ui::MainWindow)
 {
+    // Setup translations
+    appTranslator.load(QLocale::system().name(), QStringLiteral(":/i18n/i18n/"));
```
Comment (Vogtinator) [inline mainwindow.cpp]: Drop this here and do it in `main.cpp` again, basically revert the `main.cpp` changes.

Having two translators does not hurt, they are queried in reverse order, which is the desired behaviour anyway.

**Comment by Vogtinator** [comment]
> Indeed, the sidebar of the config doesn't get translated (until relaunch anyway), that's weird (probably a bug?).

Probably. No idea why - I guess the `ConfigPagesModel` doesn't keep bindings alive.
I'll try to find a workaround.

> Regarding the saving feature, it's already the case, actually :)

Indeed, I just didn't notice it...

File: mainwindow.cpp (hunk 3 above)
Comment (adriweb) [reply mainwindow.cpp]: Can't happen ; all the rest is done.

File: mainwindow.ui (hunk 6 above)
Comment (adriweb) [reply mainwindow.ui]: done

File: mainwindow.h (hunk 5 above)
Comment (adriweb) [reply mainwindow.h]: done

File: mainwindow.cpp (hunk 4 above)
Comment (adriweb) [reply mainwindow.cpp]: done

File: mainwindow.cpp (hunk 7 above)
Comment (adriweb) [reply mainwindow.cpp]: done

File: mainwindow.cpp [hunk 8]
```diff
@@ -110,6 +110,11 @@ MainWindow::MainWindow(QWidget *parent) :
     connect(ui->actionAbout_Firebird, SIGNAL(triggered(bool)), this, SLOT(showAbout()));
     connect(ui->actionAbout_Qt, SIGNAL(triggered(bool)), qApp, SLOT(aboutQt()));
 
+    // Lang switch
```
Comment (Vogtinator) [inline mainwindow.cpp]: It seems like you can use `QLocale` to get the language names of available translations.
That would make hardcoding it here and in `mainwindow.ui` file unnecessary.

**Pull request merged**